src/twisted/test/test_sslverify.py

```diff
@@ -165,10 +167,12 @@ def certificatesForAuthorityAndServer(serviceIdentity=u'example.com'):
         L{sslverify.PrivateCertificate})
     """
     commonNameForCA = x509.Name(
-        [x509.NameAttribute(NameOID.COMMON_NAME, u'Testing Example CA')]
+        [x509.NameAttribute(NameOID.COMMON_NAME, u'Testing Example CA'),
+         x509.NameAttribute(NameOID.DOMAIN_COMPONENT, serviceIdentity)]
     )
     commonNameForServer = x509.Name(
-        [x509.NameAttribute(NameOID.COMMON_NAME, u'Testing Example Server')]
+        [x509.NameAttribute(NameOID.COMMON_NAME, u'Testing Example Server'),
+         x509.NameAttribute(NameOID.DOMAIN_COMPONENT, serviceIdentity)]
```

Comment:
I don't think that this is correct for IP addresses.

It is valid, but deprecated, to specify the domain in the common name per [RFC 2818](https://tools.ietf.org/html/rfc2818), which says:

```
   If a subjectAltName extension of type dNSName is present, that MUST
   be used as the identity. Otherwise, the (most specific) Common Name
   field in the Subject field of the certificate MUST be used. Although
   the use of the Common Name is existing practice, it is deprecated and
   Certification Authorities are encouraged to use the dNSName instead.
```

However below it says that iPAddress "must" (oddly not "MUST") be a SAN &mdash; it doesn't even mention falling back to the common name:

```
   In some cases, the URI is specified as an IP address rather than a
   hostname. In this case, the iPAddress subjectAltName must be present
   in the certificate and must exactly match the IP in the URI.
```

Furthermore, https://github.com/pyca/service_identity/pull/25 only implements iPAddress in SAN. I think that this change should be reverted &mdash; the test fixture would be more accurate without it.